docs/guide/interactions/checks.rst

```diff
@@ -0,0 +1,368 @@
+    @app_commands.checks.has_any_role('Role Name', 1234567890, 'AnotherOne')
+    async def multiple_role_check(interaction: discord.Interaction, user: discord.User):
+        await interaction.response.send_message('You have all the roles!', ephemeral=True)
+
+Let's take a quick look through the code here:
+
+- First, a simple :class:`.app_commands.Command` is registered using the :func:`@app_commands.command() <.app_commands.command>` decorator.
+- Then, a :func:`@has_role <.app_commands.checks.has_role>` check is added to ensure that a user has a given role. This role can either be a :class:`str` representing the name of the role, or an :class:`int` representing the id of the role.
+- Then, the actual function callback for the command is created.
+
+A very similar process happens for checking multiple roles:
+
+- First, a :class:`.app_commands.ContextMenu` is registered using the :func:`@app_commands.context_menu() <.app_commands.context_menu>` decorator.
+- Then, a :func:`@has_role <.app_commands.checks.has_any_role>` is added for multiple roles. Again, these roles are either :class:`str` or :class:`int` values.
+- Then, the callback for the context menu is created.
+
+.. _guide_interaction_checks_permission-check:
+
+Permission Checks
+------------------
+
+These checks handle validating whether the user activating a command has certain permissions in a server, or whether the bot user has specific permissions.
+
+.. code-block:: python3
+    :emphasize-lines: 3,9
+
+    # User permissions check.
+    @app_commands.command()
+    @app_commands.has_permissions(administrator=True)
+    async def user_permission_check(interaction: discord.Interaction):
+        await interaction.response.send_message('Hello administrator!', ephemeral=True)
+
+    # Bot permissions check.
+    @app_commands.command()
+    @app_commands.bot_has_permissions(manage_messages=True, manage_guild=True)
+    async def bot_permission_check(interaction: discord.Interaction):
+        await interaction.response.send_message('I can do that!', ephemeral=True)
+
+These two checks are identical in usage. The only difference is that one checks the permissions of the user who calls the command, and one checks the permissions of the bot user the command is issued to.
+
+The permissions given to the check can be any number of permissions, which can be found in :class:`.Permissions`, and must have a boolean value.
+
+.. _guide_interaction_checks_cooldown-check:
+
+Cooldown Checks
+----------------
+
+It is also possible to use a check decorator that will directly attach a cooldown to your command.
+
+There are two cooldown methods, static and dynamic.
+
+Static Cooldown
+~~~~~~~~~~~~~~~~
+
+Static cooldowns are methods which limit a command to a certain number of uses per time frame.
+
+.. code-block:: python3
+    :emphasize-lines: 2
+
+    @app_commands.command()
+    @app_commands.checks.cooldown(1, 5.0, key=lambda interaction: (i.guild_id, i.user.id))
```

Comment:
```suggestion
    @app_commands.checks.cooldown(1, 5.0, key=lambda interaction: (interaction.guild_id, interaction.user.id))
```
Minor syntax error here, I elected to use the change the body to `interaction` instead of the arg to `i` since I think it would be clearer to beginners